.github/workflows/test-deploy.yml

```diff
@@ -4,23 +4,31 @@ on:
     - cron: 0 0 * * *
   release:
     types: [published]
-  pull_request:
+  pull_request_target:
 jobs:
+  authorize:
+    environment:
+      ${{ (github.event_name == 'pull_request_target' &&
+      github.event.pull_request.head.repo.full_name != github.repository) &&
+      'external' || 'internal' }}
+    runs-on: ubuntu-latest
   lint:
+    needs: authorize
```

Comment:
I don't think we need to worry about linting, right?

Reply:
Linting implies running `npm ci` on the source branch, controlled by the contributor, and this could lead to controlled code execution.

Reply:
https://securitylab.github.com/research/github-actions-preventing-pwn-requests